tests/test_cuda_bd_exchange_mover.py

```diff
@@ -271,21 +271,20 @@ def test_bd_exchange_deterministic_moves(moves, precision, seed):
     if precision == np.float64:
         klass = custom_ops.BDExchangeMove_f64
 
-    # Reference that makes a single proposal per move
-    bdem_a = klass(N, group_idxs, params, DEFAULT_TEMP, nb.potential.beta, cutoff, seed, 1, 1)
+    # Reference version
+    bdem_a = klass(N, group_idxs, params, DEFAULT_TEMP, nb.potential.beta, cutoff, seed, proposals_per_move, 1)
     # Test version that makes all proposals in a single move
-    bdem_b = klass(N, group_idxs, params, DEFAULT_TEMP, nb.potential.beta, cutoff, seed, moves, 1)
+    bdem_b = klass(N, group_idxs, params, DEFAULT_TEMP, nb.potential.beta, cutoff, seed, proposals_per_move, 1)
 
-    iterative_moved_coords = conf.copy()
-    for _ in range(moves):
-        iterative_moved_coords, _ = bdem_a.move(iterative_moved_coords, box)
-        assert not np.all(conf == iterative_moved_coords)  # We should move every time since its a single mol
-    batch_moved_coords, _ = bdem_b.move(conf, box)
-    # Moves should be deterministic regardless the number of steps taken per move
-    np.testing.assert_array_equal(iterative_moved_coords, batch_moved_coords)
+    coords_a, box_a = bdem_a.move(conf, box)
```

Comment:
We are only targeting the determinism where calling `move()` produces the same number of proposals. Different numbers of proposals per `move()` call will produce different results. This was no longer deterministic since the noise was generated every `move()` which resulted in different offsets in the RNG that would produce different noise.

This is likely desirable as we can make a few more performance enhancements with this approach (avoid partitioning every proposal) and unclear what the benefit would be compared to being able to batch more moves.

Reply:
> This was no longer deterministic since the noise was generated every move() which resulted in different offsets in the RNG that would produce different noise.

I'm not sure I understand this - is it possible to clarify? It seems like there would be a 1-1 correspondence between offset in the RNG and proposal regardless of `proposals_per_move`, unless the RNG state is shared with code outside of `move` (is this the case?)

Reply:
Realized that this was not to do with the offsets, rather the 'mixing' of what was previously sequential noise.

In the previous implementation we would generate some fixed quantity of noise upfront in a batch size, which would be consumed with every proposal, and could be used between multiple calls to `move`.  In the new implementation, we simply generate all the noise needed for that specific `move()` call, importantly with multiple calls to the same RNG. There is additional nuance around the device level API for translations in the targeted case, but going to ignore that.

Before we would generate 1000 values for the translations, 1000 values for rotations. This is two calls to the RNG and leaves us with an RNG with an offset of 2000. Regardless of the number of proposals, we would get the same noise for each subsequent step. Proposal zero gets you translations[0:10] and rotations[0:10], proposal one translation[10:20], etc...

In the new implementation if you have one proposal per move of one you get 10 translations and 10 rotations, if you have two proposals per move you get 20 translations and 20 rotations. The noise that was previously split between translations (first ten) and rotations (second ten) are now all translations it won't produce the same result. And note that it would be difficult to mix them into a single array because rotations are from the normal distribution and translations are uniform.

Reply:
Ah, I think I understand, thanks for clarifying.

> The noise that was previously split between translations (first ten) and rotations (second ten) are now all translations it won't produce the same result

Would it be possible to avoid this issue by consuming the samples in a different order? E.g., using even samples for translations and odd for rotations?

Reply:
> And note that it would be difficult to mix them into a single array because rotations are from the normal distribution and translations are uniform.

OK, I see.

One other thought - would it make sense to use separate RNG state instances for translations and rotations? Seems like it might be worth it to have the property of invariance with respect to `proposals_per_move`.

Reply:
> Would it be possible to avoid this issue by consuming the samples in a different order? E.g., using even samples for translations and odd for rotations?

No as they are different types of random values, uniform for translations and normal for rotations.

> would it make sense to use separate RNG state instances for translations and rotations?

That might be the thing to do, there is probably some nuance with the device API but as long as we do proposals per move that are factors of 128 (the number of device RNGs) we should be good.

Reply:
(looks like this was addressed in https://github.com/proteneer/timemachine/pull/1234/commits/e162ec2e947fc767e72ebab34867d2f1cb880e45)